proc-macros/src/render_client.rs

```diff
@@ -109,10 +105,20 @@ impl RpcDescription {
 			let ret_ty = args.last_mut().unwrap();
 			let err_ty = self.jrps_client_item(quote! { core::client::Error });
 
+			quote! { core::result::Result<#ret_ty, #err_ty> }
+		} else if type_name.ident == "ResponsePayload" {
```

Comment:
I'm sad that we have to match on the string type names :D

I guess it wasn't possible to use a trait to convert these into results automatically or something; I'm sure it was something like that but can't remember the issue now! But ah well!

Reply:
Nah, it's that the client codegen is super opinionated and if the error type must be `jsonrpsee::core::client::Error` otherwise it won't compile. Yeah, it's ugly but just proc macro code thank god.

IntoResponse stuff doesn't work there because we have deal with I/O errors and bunch of things